File: PraiseAnimation/Classes/PraiseAnimation.m
//
//  PraiseAnimation.h
//  PraiseAnimation
//
//  Created by Derrick on 2020/9/25.
//  Copyright © 2020 Victor. All rights reserved.
//

#import "PraiseAnimation.h"
#import "PraiseImage.h"
@interface PraiseAnimation()
@property (nonatomic ,strong) NSArray<UIImage *>* imageArray;
@property (nonatomic ,weak) UIView *view;

@end
@implementation PraiseAnimation
-(instancetype)initWithImageArray:(NSArray<UIImage *>*)imageArray onView:(UIView *)view point:(CGPoint)point{
    self = [super init];
    if (self) {
        _imageArray = imageArray;
        _view = view;
        _point = point;
        _imageSize = CGSizeMake(25, 25);
        _speed = 5;
        _animationH = 300.0;
        _leftSpacing = 50.0;
        _rightSpacing = 50.0;
    }
    return self;
}

- (instancetype)initWithImageArray:(NSArray<UIImage *> *)imageArray onView:(UIView *)view {
    self = [super init];
    if (self) {
        _imageArray = imageArray;
        _view = view;
        _imageSize = CGSizeMake(25, 25);
        _speed = 5;
        _animationH = 300.0;
        _leftSpacing = 50.0;
        _rightSpacing = 50.0;
    }
    return self;
}
-(void)animate:(NSInteger)count{
    if (count == 0) count = 1;
    for (int i = 0; i < count; i++) {
        int random = (int)[_imageArray count];
        NSInteger index = arc4random_uniform(random);
        UIImage *image = _imageArray[index];
        PraiseImage *praise = [[PraiseImage alloc] initWithImage:image];
        praise.frame = CGRectMake(_point.x, _point.y, _imageSize.width, _imageSize.height);
        praise.animateH = _animationH;
        praise.speed = _speed;
        praise.minX = _point.x - _leftSpacing;
        praise.maxX = CGRectGetMaxX(praise.frame) + _rightSpacing;
        [_view addSubview:praise];
        [praise animate];
    }
}


@end

File: PraiseAnimation.swift
//
//  PraiseAnimation.swift
//  PopTipSwift_Example
//
//  Created by VICTOR03 on 2021/5/19.
//  Copyright © 2021 CocoaPods. All rights reserved.
//

import UIKit

open class PraiseAnimation {
    /// 动画对象大小 默认(25,25)
    public var imageSize = CGSize(width: 25, height: 25)
    /// 动画移动高度 默认(300)
    public var animationH:CGFloat = 300
    /// 动画左右幅度 以起点位置为参照点 默认50
    public var leftSpacing:CGFloat = 50
    public var rightSpacing:CGFloat = 50
    /// 动画速度 默认 5
    public var speed:Int = 5
    /// 动画位置
    public var point:CGPoint = .zero
    
    private var imageArray:[UIImage] = []
    private var view:UIView!
    
    public init(with imageArray:[UIImage], onView: UIView, point:CGPoint) {
        self.imageArray = imageArray
        self.view = onView
        self.point = point
    }
  
    public func animate(count:Int = 1) {
        for _ in 0..<count {
            let random = imageArray.count
            let index = Int(arc4random_uniform(UInt32(random)))
            let image = imageArray[index]
            let praise = PraiseImageView()
            praise.image = image
            praise.frame = CGRect(x: point.x,
                                  y: point.y,
                                  width: image.size.width,
                                  height: image.size.height)
            praise.animateH = animationH
            praise.speed = speed
            praise.minX = point.x - leftSpacing
            praise.maxX = praise.frame.maxX + rightSpacing
            view.addSubview(praise)
            praise.animate()
        }
    }
}
